libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs

```diff
@@ -4,70 +4,38 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Net;
-using System.Net.Http;
 using System.Security.Cryptography;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Bot.Schema;
-using Newtonsoft.Json;
 
 namespace Microsoft.Bot.Builder.Adapters.Facebook
 {
     public class FacebookAdapter : BotAdapter
     {
         public const string NAME = "Facebook Adapter";
 
-        private readonly IFacebookAdapterOptions options;
+        private readonly FacebookAdapterOptions _options;
 
-        public FacebookAdapter(IFacebookAdapterOptions options)
+        public FacebookAdapter(FacebookAdapterOptions options)
         {
-            this.options = options;
-            this.options.ApiHost = "graph.facebook.com";
-            this.options.ApiVersion = "v3.2";
+            _options = options;
+            _options.ApiHost = "graph.facebook.com";
+            _options.ApiVersion = "v3.2";
 
-            if (string.IsNullOrEmpty(this.options.AccessToken) && this.options.GetAccessTokenForPageAsync != default(Func<string, Task<string>>))
+            if (string.IsNullOrEmpty(_options.AccessToken) && _options.GetAccessTokenForPageAsync != default(Func<string, Task<string>>))
```

Comment:
Also, here we are validation that options has an AccessToken, but we're not checking first if options is null or not. Add a validation which throws an ArgumentNullException for options (and for the Facabook API parameter after you add it, as @ceciliaavila mentioned above)